src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -234,43 +243,88 @@ public synchronized boolean saveConfig() {
     }
 
     /**
-     * Copies a file
+     * Create directory provided from the parameter
+     *
+     * @param dirPath - directory to be created
      *
-     * @param file
-     *            the file to copy from
-     * @param destFile
-     *            the file to copy to
+     * @return
+     *  True if directory was created successfully or directory already existed
+     *  False if directory was failed to create
      */
-    private void copyFile(File file, File destFile) {
-        BufferedInputStream in = null;
-        BufferedOutputStream out = null;
-        try {
-
-            // reading and writing 8 KB at a time
-            int bufferSize = 8 * 1024;
+    private static boolean createDir(File dirPath) {
+        boolean isSuccessful = true;
+        if (!dirPath.exists() || !dirPath.isDirectory()) {
+            isSuccessful = dirPath.mkdirs();
+            if (isSuccessful) {
+                logger.info("Created config directory: " + dirPath);
+            } else {
+                logger.info("Unable to create config directory: " + dirPath);
+            }
+        } else {
+            logger.info("Config directory already exists: " + dirPath);
+        }
+        return isSuccessful;
+    }
 
-            byte[] buf = new byte[bufferSize];
+    /**
+     * Returns current user's Dataloader configuration directory
+     *  ie) For Windows - C:\Users\{user}\AppData\Local\salesforce.com\Data Loader {version}\conf
+     *      For Mac - /Users/{user}/Library/Preferences/salesforce.com/Data Loader {version}/conf
+     *      For *nix - ~/.salesforce.com/Data Loader {version}/conf
+     *
+     * @return
+     *  Current user's Dataloader configuration directory
+     */
+    private static File getUserConfigDir() {
+        File dir;
 
-            in = new BufferedInputStream(new FileInputStream(file), bufferSize);
-            int r = 0;
+        switch (OS_TYPE) {
+            case WINDOWS: {
+                dir = Paths.get(System.getProperty("user.home"), "AppData/Local", APP_VENDOR, getProductName(), CONFIG_DIR).toFile();
```

Comment:
While this will most likely work (Win32 tries to be compatible when it can, and Java on Windows benefits from that), you're using the wrong character for a Windows path separator here.